research/encoding_2d_data/encode.py

```diff
@@ -0,0 +1,36 @@
+import os
```

Comment:
Please add some text to encode.py explaining how data is being encoded, without having to read the entire code

Reply:
(this could/should be done by updating the readme explaining the encoding/decoding method to get rid of rotations)